### PR TITLE
sys_usbd tidy ups

### PR DESCRIPTION
**Pull request by bevanweiss**
Tidy up fake transfer iterator handling. erase invalidates all iterators including the current iterator (i.e. 'it'), given precedence ordering this was UB prior to C++17.  Splitting out to use return iterator from erase seems cleaner.
Also added some additional info to usb debug message to potentially help with #8666, and used the atomic (dev_counter) less often

### CODE REVIEW

File: rpcs3/Emu/Cell/lv2/sys_usbd.cpp [hunk 1]
```diff
@@ -387,14 +390,13 @@ void usb_handler_thread::check_devices_vs_ldds()
 					continue;
 				}
 
-				sys_usbd.success("Ldd device matchup for <%s>", ldd.name);
-
 				dev->read_descriptors();
+				dev->assigned_number = dev_counter++; // assign current dev_counter, and atomically increment
+
+				sys_usbd.success("Ldd device matchup for <%s>, assigned as handled_device=0x%x", ldd.name, dev->assigned_number);
```
Comment (Megamouse) [inline rpcs3/Emu/Cell/lv2/sys_usbd.cpp:396]: Why is this printed as hex? It's just some counter which is humanly readable as a regular number

Comment (bevanweiss) [reply rpcs3/Emu/Cell/lv2/sys_usbd.cpp:396]: It seems that sys_usbd.cpp only uses HEX display for everything... like:
L495: ... max_devices=0x%x...
L696: ... buf_length=0x%x....
L699: ...., wIndex:0x%x, wLength:0x%x" ....
L787:  .... count=*0x%x)"....  <-- ok, this one is a little strange

So I didn't want to break that consistency just for my change.
If the plain old decimal format is preferred, I've got no issue with that.

Comment (Megamouse) [reply rpcs3/Emu/Cell/lv2/sys_usbd.cpp:396]: whatever you want

File: rpcs3/Emu/Cell/lv2/sys_usbd.cpp [hunk 2]
```diff
@@ -173,7 +173,10 @@ usb_handler_thread::usb_handler_thread()
 		check_device(0x1415, 0x0000, 0x0000, "Singstar Microphone");
 		check_device(0x12BA, 0x0100, 0x0100, "Guitar Hero Guitar");
 		check_device(0x12BA, 0x0120, 0x0120, "Guitar Hero Drums");
-		found_ghltar = check_device(0x12BA, 0x074B, 0x074B, "Guitar Hero Live Guitar");
+		if (check_device(0x12BA, 0x074B, 0x074B, "Guitar Hero Live Guitar"))
```
Comment (Megamouse) [inline rpcs3/Emu/Cell/lv2/sys_usbd.cpp:176]: And what exactly is incorrect?

Comment (Megamouse) [reply rpcs3/Emu/Cell/lv2/sys_usbd.cpp:176]: I see

Comment (Megamouse) [reply rpcs3/Emu/Cell/lv2/sys_usbd.cpp:176]: Good catch

**Pull request merged**